api/src/main/java/org/openmrs/module/patientflags/api/impl/FlagServiceImpl.java

```diff
@@ -224,7 +233,19 @@ public void purgeFlag(Integer flagId) {
 		// then refresh the cache
 		refreshCache();
 	}
-	
+
+	public void retireFlag(Flag flag, String retiredReason) {
+		if (retiredReason != null && retiredReason.length() >= 1) {

```

Comment:
Do you allow empty spaces for the retire reason? Shouldn't you be using StringUtils.IsNotBlank()?